salt/client/__init__.py

```diff
@@ -248,7 +248,7 @@ def _check_pub_data(self, pub_data):
         '''
         Common checks on the pub_data data structure returned from running pub
         '''
-        if not pub_data:
+        if pub_data == '':
```

Comment:
One will match specifically against an empty string, the other will match against non truthy-things, like an empty list/dictionary.